### PR TITLE
[Feat]: Added timer

### PR DESCRIPTION
**Pull request by aryan-13**
Added:
- Timer Context
- Start Timer
- Pause Timer
- Reset Timer
- Switch between Work and Break timer

image

### CODE REVIEW

**Review by prachi-sahani — COMMENTED** [review]
All the changes look fine and are good to merge
suggestion - consider formatting your code

File: src/Components/MainSection/TaskList/TaskList.jsx [hunk 1]
```diff
@@ -20,14 +21,16 @@ function TaskList() {
...
 			<button
-				className="btn"
+				className="btn btn-grey"
 				onClick={() => {
 					tasklistDispatch({
 						type: 'ADD_TASK',
 						payload: { id: uuidv4(), name: task, isCompleted: false },
 					});
+					setTask('');
 				}}
```
Comment (JantuDeb) [inline src/Components/MainSection/TaskList/TaskList.jsx:34]: create another function for this

File: src/Components/Timer/Timer.jsx [hunk 2]
```diff
@@ -1,6 +1,42 @@
...
+					}}
+				>
+					Break
+				</button>
+			</div>
+			{timerWork ? (
+				<Clock key={tempKey1} setKey={setTempKey1} type={'work'} />
+			) : (
+				<Clock key={tempKey2} setKey={setTempKey2} type={'break'} />
+			)}
```
Comment (JantuDeb) [inline src/Components/Timer/Timer.jsx:37]:
```js
<Clock  type={timerWork ? 'work':'break'} />
```
You can do something like this which will eliminate the need of a `key`

**Review by JantuDeb — APPROVED** [review]
Overall the code looks good, I left few comments you can check this out

File: src/Components/Timer/Timer.jsx (hunk 2 above)
Comment (aryan-13) [reply src/Components/Timer/Timer.jsx:37]: This was not working without key

File: src/Components/MainSection/TaskList/TaskList.jsx (hunk 1 above)
Comment (aryan-13) [reply src/Components/MainSection/TaskList/TaskList.jsx:34]: Yes changed

**Pull request merged**